tpm2_pytss/internal/utils.py

```diff
@@ -245,7 +245,9 @@ def _lib_version_atleast(tss2_lib, version):
     if tss2_lib not in _versions:
         return False
     # Needed to remove git commit from version strings
-    lib_parts = _versions[tss2_lib].rsplit("-", 1)
+    # for example 3.1.0-126-g0fd1c5fbbaf2 should be
+    # 3.1.0
+    lib_parts = _versions[tss2_lib].rsplit("-")
```

Comment:
The second part is sometimes needed, for example I use it in the tests for the tools encoder/decoder as some new YAML output isn't in a released version yet.
There is also RC versions, 3.3.0 is not the same as 3.3.0-rc0 and we might need to keep track of that

Reply:
I updated the PR and included an internal test case with version strings to match to expected outputs.